xds/internal/resolver/xds_resolver_test.go

```diff
@@ -1178,295 +1265,349 @@ func (s) TestXDSResolverMaxStreamDuration(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			req := iresolver.RPCInfo{
 				Method:  tc.method,
-				Context: context.Background(),
+				Context: ctx,
 			}
 			res, err := cs.SelectConfig(req)
 			if err != nil {
-				t.Errorf("Unexpected error from cs.SelectConfig(%v): %v", req, err)
+				t.Errorf("cs.SelectConfig(%v): %v", req, err)
 				return
 			}
 			res.OnCommitted()
 			got := res.MethodConfig.Timeout
-			if !reflect.DeepEqual(got, tc.want) {
+			if !cmp.Equal(got, tc.want) {
```

Comment:
What is reason for this switch? (Not questioning correctness, just wondering because I spend a tonnnnn of time getting cmp.Equal() to work in my OpenCensus instrumentation e2e tests).

Reply:
Cool thanks.